packages/core/src/config/parse.ts

```diff
@@ -1817,14 +1876,195 @@ const assertValidContractVariables = (
   return parsedVariables
 }
 
+const isContractReferenceLike = (value: string) => {
+  if (
+    typeof value === 'string' &&
+    value.startsWith('{{') &&
+    value.endsWith('}}')
+  ) {
+    return true
+  } else {
+    return false
+  }
+}
+
+// Borrowed from the Hardhat ethers plugin. Unfortunately, it does not export this logic.
+// So since we had to copy this function into our code base anyway, we've also split the
+// validation logic out into this function so it can be performed during the config validation
+// step.
+// Source: https://github.com/NomicFoundation/hardhat/blob/5426f582c28f90576300378c94a86e61ca13bc7b/packages/hardhat-ethers/src/internal/helpers.ts#L163
+const validateContractLibraries = async (
```

Comment:
The original hardhat logic actually was a single function. I intentionally split it into two functions because it dramatically simplifies things. 

 > We could follow the same pattern as assertValidConstructorArgs, which returns parsedConstructorArgs

We can't do that because the library address is injected into the artifact bytecode which is not stored on the parsed config.

I'm happy to explain more on a call if you want, there are quite a few reasons why merging these functions is a bad idea.

Reply:
> > We could follow the same pattern as assertValidConstructorArgs, which returns parsedConstructorArgs
> 
> We can't do that because the library address is injected into the artifact bytecode which is not stored on the parsed config.

I meant we could follow the general pattern of validating and parsing at the same time. Not that we should include it on the parsed config.

> I'm happy to explain more on a call if you want, there are quite a few reasons why merging these functions is a bad idea.

Yeah could we hop on a call? It's just unclear why splitting the logic simplifies things